python/telemetry/ctim.py

```diff
@@ -0,0 +1,365 @@
+                                          -0.000000002361], Int16ub),
+    'sw_ana_eps_3p3_ref' / PolynomialAdapter([0.0, 0.001611], Int16ub),
+    'sw_ana_eps_bus_i' / PolynomialAdapter([0.0, 0.001221], Int16ub),
+    'sw_ana_xact_i' / PolynomialAdapter([-0.02489, 0.001992], Int16ub),
+    'sw_ana_uhf_i' / PolynomialAdapter([-0.02489, 0.001992], Int16ub),
+    'sw_ana_sband_i' / PolynomialAdapter([-0.02489, 0.001992], Int16ub),
+    'sw_ana_inst_i' / PolynomialAdapter([0.0, 0.000807], Int16ub),
+    'sw_ana_hk_3p3_ref' / PolynomialAdapter([0.0, 0.001611], Int16ub),
+    'sw_ana_ifb_i' / PolynomialAdapter([0.0, 0.000269], Int16ub),
+    'sw_ana_ifb_temp' / PolynomialAdapter([98.0, -0.05936, 1.641e-05,
+                                          -0.000000002361], Int16ub),
+    'sw_adcs_eclipse' / Int8ul,
+    'adcs_info' / adcs_info,
+    'sw_adcs_analogs_voltage_2p5' / PolynomialAdapter([0.0, 0.015], Int8ul),
+    'sw_adcs_analogs_voltage_1p8' / PolynomialAdapter([0.0, 0.015], Int8ul),
+    'sw_adcs_analogs_voltage_1p0' / PolynomialAdapter([0.0, 0.015], Int8ul),
+    'sw_adcs_analogs_det_temp' / PolynomialAdapter([0.0, 0.8], Int8sl),
+    'sw_adcs_analogs_motor1_temp' / PolynomialAdapter([0.0, 0.005], Int16sb),
+    'sw_adcs_analogs_motor2_temp' / PolynomialAdapter([0.0, 0.005], Int16sb),
+    'sw_adcs_analogs_motor3_temp' / PolynomialAdapter([0.0, 0.005], Int16sb),
+    'spare_16' / Int16sl,
+    'sw_adcs_analogs_digital_bus_v' / PolynomialAdapter([0.0, 0.00125],
+                                                        Int16sb),
+    'sw_adcs_cmd_acpt' / Int8ul,
+    'sw_adcs_cmd_fail' / Int8ul,
+    'sw_adcs_time' / Int32sl,
+    'sw_adcs_sun_vec1' / PolynomialAdapter([0.0, 0.0001], Int16sb),
+    'sw_adcs_sun_vec2' / PolynomialAdapter([0.0, 0.0001], Int16sb),
+    'sw_adcs_sun_vec3' / PolynomialAdapter([0.0, 0.0001], Int16sb),
+    'sw_adcs_wheel_sp1' / PolynomialAdapter([0.0, 0.4], Int16sb),
+    'sw_adcs_wheel_sp2' / PolynomialAdapter([0.0, 0.4], Int16sb),
+    'sw_adcs_wheel_sp3' / PolynomialAdapter([0.0, 0.4], Int16sb),
+    'sw_adcs_body_rt1' / PolynomialAdapter([0.0, 5E-9], Int32sb),
+    'sw_adcs_body_rt2' / PolynomialAdapter([0.0, 5E-9], Int32sb),
+    'sw_adcs_body_rt3' / PolynomialAdapter([0.0, 5E-9], Int32sb),
+    'sw_adcs_body_quat1' / PolynomialAdapter([0.0, 5E-10], Int32sb),
+    'sw_adcs_body_quat2' / PolynomialAdapter([0.0, 5E-10], Int32sb),
+    'sw_adcs_body_quat3' / PolynomialAdapter([0.0, 5E-10], Int32sb),
+    'sw_adcs_body_quat4' / PolynomialAdapter([0.0, 5E-10], Int32sb),
+    'sw_spare_0' / Int32ul,
+    'sw_spare_1' / Int32ul,
+    'sw_spare_2' / Int32ul,
+    'sw_spare_3' / Int32ul
+)
+
+
+ctim = Struct(
+    'ax25_header' / Header,
+    'primary_header' / ccsds_space_packet.PrimaryHeader,
+    'secondary_header' / If(
+        lambda c: c.primary_header.secondary_header_flag,
+        SecondaryHeader
+    ),
+    'packet' / Switch(
+        lambda c: (c.primary_header.AP_ID),
```

Comment:
The parenthesis around  `c.primary_header.AP_ID` are not necessary. Consider removing them.